### PR TITLE
Make `DraggableScrollableController` a `ChangeNotifier`

### LINKED ISSUE #96088: Make `DraggableScrollableController` a `ChangeNotifier`
It'd be nice to be able to call `myDraggableController.addListener...` like I already can with `myScrollController` and other similar controller classes.

### PR DESCRIPTION
**Pull request by caseycrogers**
This PR allows developers to add listeners on `DraggableScrollableController` that fire whenever an attached sheet's size changes. Listening for changes is already possible via `DraggableScrollableNotification`. This PR is still useful because:
1) Depending on the situation, it is often much easier to register a listener on a controller than by using a notification listener. See: https://github.com/ueman/feedback/blob/642bcea820708cf0490a31209c928c03e04c6b2d/feedback/lib/src/feedback_widget.dart#L61
2) Other controllers (such as scroll controller) are change notifiers so developers will expect the same form `DraggableScrollableController`.

Fixes #96088

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart [hunk 1]
```diff
@@ -44,7 +44,7 @@ typedef ScrollableWidgetBuilder = Widget Function(
 ///
 /// The controller's methods cannot be used until after the controller has been
 /// passed into a [DraggableScrollableSheet] and the sheet has run initState.
-class DraggableScrollableController {
+class DraggableScrollableController extends ChangeNotifier {
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart:47]: Please document what you can listen for.

File: packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart [hunk 2]
```diff
@@ -160,9 +160,11 @@ class DraggableScrollableController {
   void _attach(_DraggableScrollableSheetScrollController scrollController) {
     assert(_attachedController == null, 'Draggable scrollable controller is already attached to a sheet.');
     _attachedController = scrollController;
+    _attachedController!.extent.addListener(notifyListeners);
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart:163]: Somehow you'll have to move the listener over if the extent gets replaced via `_replaceExtent`.

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart:163]: (Please add a test that covers this case)

Comment (caseycrogers) [reply packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart]: Good catch! Fixed behaviors and added tests.

File: packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart (hunk 1 above)
Comment (caseycrogers) [reply packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart:54]: Added. Let me know if anything more is needed.

File: packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart [hunk 3]
```diff
@@ -976,6 +976,148 @@ void main() {
...
+      null,
+      minChildSize: .4,
+      controller: controller,
+      stackKey: stackKey,
+      containerKey: containerKey,
+    ));
+    expect(loggedSizes, <double>[.4].map((double v) => closeTo(v, precisionErrorTolerance)));
+    loggedSizes.clear();
+  });
+
```
Comment (goderbauer) [inline packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart:1119]: nit: remove extra blank line

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
@dnfield Can you do a secondary review?

**Review by dnfield — APPROVED** [review]

**Comment by pedrohsampaioo** [comment]
@caseycrogers do you know when this change will be available in the stable? I need listener the DraggableScrollableSheet size changes...

**Comment by caseycrogers** [comment]
> @caseycrogers do you know when this change will be available in the stable? I need listener the DraggableScrollableSheet size changes...

Sadly I don't know when this will make it into stable-I'm honestly quite surprised it hasn't already made it in.

While you're waiting for this to go in you can always use `NotificationListener<DraggableScrollableNotification>...`. It's not as elegant but it should work just fine.

**Comment by pedrohsampaioo** [comment]
> > @caseycrogers do you know when this change will be available in the stable? I need listener the DraggableScrollableSheet size changes...
> 
> Sadly I don't know when this will make it into stable-I'm honestly quite surprised it hasn't already made it in.
> 
> While you're waiting for this to go in you can always use `NotificationListener<DraggableScrollableNotification>...`. It's not as elegant but it should work just fine.

Ok, thanks I'm doing something like you said

**Pull request merged**